src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MIN_SIZE = 1;
+    private static final int MAX_SIZE = 9;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private Figure square = new Square(random.nextInt(MIN_SIZE, MAX_SIZE),
+            new ColorSupplier().getRandomColor());
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(random.nextInt(MIN_SIZE, MAX_SIZE),
+            random.nextInt(MIN_SIZE, MAX_SIZE), random.nextInt(MIN_SIZE, MAX_SIZE),
+            new ColorSupplier().getRandomColor());
+    private Figure rightTriangle = new RightTriangle(random.nextInt(MIN_SIZE, MAX_SIZE),
+            random.nextInt(MIN_SIZE, MAX_SIZE),
+            new ColorSupplier().getRandomColor());
+    private Figure circle = new Circle(random.nextInt(MIN_SIZE, MAX_SIZE),
+            new ColorSupplier().getRandomColor());
+    private Figure rectangle = new Rectangle(random.nextInt(MIN_SIZE, MAX_SIZE),
+            random.nextInt(MIN_SIZE, MAX_SIZE),
+            new ColorSupplier().getRandomColor());
+    private Figure[] figures = new Figure[] {rectangle, circle, rightTriangle,
+            isoscelesTrapezoid, square};
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(figures.length);
+        return figures[randomIndex];
+    }
```

Comment:
Please rework this solution.
Now you have 5 predefined figures. So if you need 100 random figures you'll get about 20 duplicates of each figure.
So it not really works like random.

During method execution you should firstly to generate random number to determine shape you need and then generate this shape with random properties
So if you create 100 random figures you anyway will have about 20 squares but all of them will be different in color and size